tests/testthat/test-biology-vivax.R

```diff
@@ -0,0 +1,250 @@
+                               list(human_population = population))
+  parameters <- set_equilibrium(parameters, EIR)
+
+  eq <- malariaEquilibriumVivax::vivax_equilibrium(
+    age = EQUILIBRIUM_AGES,
+    ft = 0,
+    EIR = EIR,
+    p = translate_vivax_parameters(parameters))$states
+
+  variables <- create_variables(parameters)
+  het <- statmod::gauss.quad.prob(parameters$n_heterogeneity_groups, dist = "normal")
+  expect_equal(
+    mean(
+      clinical_immunity(
+        variables$ica$get_values(),
+        variables$icm$get_values(),
+        parameters
+      )
+    ),
+    sum(unlist(lapply(1:parameters$n_heterogeneity_groups, function(h){
+      weighted.mean(eq[[h]][,'phi_clin'], eq[[h]][,'prop'])}
+      )) * het$weights),
+    tolerance=1e-2
+  )
+})
+
+test_that('mosquito_limit is set to 0 for 0 EIR', {
+  parameters <- parameterise_mosquito_equilibrium(get_parameters(parasite = "vivax", list(
+    individual_mosquitoes = TRUE
+  )), 0)
+  expect_equal(parameters$mosquito_limit, 0)
+})
+
+test_that('mosquito_limit is set to a sensible level', {
+  EIRs <- 5
+
+  seasonalities <- list(
+    list(
+      g0 = 7.2,
+      g = c(-10, 4.06, -1.08),
+      h = c(-4.6, 3.59, -1.32)
+    ),
+    list(
+      g0 = 7.2,
+      g = c(-10, 4.06, -1.08),
+      h = c(-4.6, 3.59, -1.32)
+    )
+  )
+
+  for (EIR in EIRs) {
+    for (seasonality in seasonalities) {
+      parameters <- get_parameters(parasite = "vivax", c(
+        seasonality,
+        model_seasonality = TRUE,
+        init_foim = .1
+      ))
+      parameters <- parameterise_mosquito_equilibrium(parameters, EIR)
+      run_simulation(5, parameters)
+    }
+  }
+  expect_true(TRUE)
+})
+
+test_that('mosquito_effects correctly samples mortalities and infections without interventions', {
+  parameters <- get_parameters(parasite = "vivax")
+  events <- create_events(parameters)
+  variables <- create_variables(parameters)
+  infected <- individual::Bitset$new(100)$insert(1:25)
+  dead <- individual::Bitset$new(100)$insert(c(1:15, 51:60, 76:85))
+  bernoulli_mock = mockery::mock(infected, dead)
+  mockery::stub(biting_effects_individual, 'sample_bitset', bernoulli_mock)
+  events$mosquito_infection <- mock_event(events$mosquito_infection)
+  events$mosquito_death <- mock_event(events$mosquito_death)
+  variables$mosquito_state <- mock_category(
+    c('Sm', 'Pm', 'Im', 'NonExistent'),
+    rep('Sm', 100)
+  )
+  biting_effects_individual(
+    variables,
+    .5,
+    events,
+    1,
+    individual::Bitset$new(100)$insert(1:50),
+    individual::Bitset$new(100)$insert(1:100),
+    parameters$mum,
+    parameters,
+    timestep = 1
+  )
+
+  mockery::expect_args(
+    variables$mosquito_state$queue_update_mock(),
+    1,
+    'Pm',
+    infected
+  )
+
+  expect_equal(
+    mockery::mock_args(bernoulli_mock)[[2]][[2]],
+    parameters$mum
+  )
+
+  mockery::expect_args(
+    events$mosquito_death$schedule,
+    1,
+    dead,
+    0
+  )
+})
```

Comment:
These tests don't depend on falciparum or vivax right? We can just remove them

Reply:
Good shout! Removed in the next commit.